lib/foreman_discovery/engine.rb

```diff
@@ -78,7 +78,7 @@ class Engine < ::Rails::Engine
               full_name: N_("Hostname facts"),
               description: N_("List of facts to use for the hostname (first wins)")
 
-            validates "discovery_hostname", presence: true
+            # validates "discovery_hostname", presence: true
```

Comment:
We had a discussion about this in https://github.com/theforeman/foreman_discovery/pull/612#discussion_r1450829762. For now I wouldn't change it.

Reply:
https://github.com/theforeman/foreman_discovery/pull/622 has an alternative approach.